src/main/java/com/exasol/adapter/dialects/exasol/ExasolColumnMetadataReader.java

```diff
@@ -37,22 +53,102 @@ public ExasolColumnMetadataReader(final Connection connection, final AdapterProp
     }
 
     @Override
-    public DataType mapJdbcType(final JDBCTypeDescription jdbcTypeDescription) {
+    public DataType mapJdbcType(final JDBCTypeDescription type) {
+        final DataType resultType = map(type);
+        LOGGER.fine(() -> "Mapped JDBC type '" + type.getTypeName() + "' / " + type.getJdbcType() + " with byte size "
+                + type.getByteSize() + ", decimal scale " + type.getDecimalScale() + ", precision/size "
+                + type.getPrecisionOrSize() + " to " + resultType);
+        return resultType;
+    }
+
+    private DataType map(final JDBCTypeDescription jdbcTypeDescription) {
+        DataType type = createTypeBasedOnTypeName(jdbcTypeDescription);
+        if (type != null) {
+            return type;
+        }
+        type = createTypeBasedOnJdbcType(jdbcTypeDescription);
+        if (type != null) {
+            return type;
+        }
+
+        return super.mapJdbcType(jdbcTypeDescription);
+    }
+
+    private DataType createTypeBasedOnTypeName(final JDBCTypeDescription jdbcTypeDescription) {
+        if (jdbcTypeDescription.getTypeName() == null) {
+            return null;
+        }
+        switch (jdbcTypeDescription.getTypeName()) {
+        case TYPE_NAME_HASHTYPE:
+            return createHashtypeType(jdbcTypeDescription);
+        case TYPE_NAME_GEOMETRY:
+            return createGeometryType(jdbcTypeDescription);
+        case TYPE_NAME_INTERVAL_YEAR_TO_MONTH:
+            return createIntervalYearToMonthType(jdbcTypeDescription);
+        case TYPE_NAME_INTERVAL_DAY_TO_SECOND:
+            return createIntervalDayToSecondType(jdbcTypeDescription);
+        case TYPE_NAME_TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+            return DataType.createTimestamp(true);
+        case TYPE_NAME_TIMESTAMP:
+            return DataType.createTimestamp(false);
+        default:
+            return null;
+        }
+    }
+
+    private DataType createTypeBasedOnJdbcType(final JDBCTypeDescription jdbcTypeDescription) {
         switch (jdbcTypeDescription.getJdbcType()) {
-        case EXASOL_INTERVAL_DAY_TO_SECONDS:
-            return DataType.createIntervalDaySecond(jdbcTypeDescription.getPrecisionOrSize(),
-                    jdbcTypeDescription.getDecimalScale());
-        case EXASOL_INTERVAL_YEAR_TO_MONTHS:
-            return DataType.createIntervalYearMonth(jdbcTypeDescription.getPrecisionOrSize());
+        case EXASOL_HASHTYPE:
```

Comment:
The issue sais you wanted to use `getEXAType` but I could not find it in the PR. Is that intended?

Reply:
Yes. getEXAType() is not yet accessible in JDBC driver 7.1.3. We need to wait for 7.1.4. Only then we can really fix #60